design/one-pager-host-aware-stack-manager.md

```diff
@@ -0,0 +1,211 @@
+# Host Aware Stack Manager
```

Comment:
It looks like you've included #1138 in this PR.  I don't think we need to include it here since it doesn't provide any code changes and design changes can be tackled separately.

Reply:
Agreed, done!